pkg/bootkube/bootkube.go

```diff
@@ -129,13 +130,27 @@ func (b *bootkube) Run() error {
 			errch <- err
 		}
 	}()
+	var etcdServiceIP string
+	var err error
 	if b.selfHostedEtcd {
-		requiredPods = append(requiredPods, "etcd-operator")
+		etcdServiceIP, err = detectEtcdIP(b.assetDir)
```

Comment:
move this into the block below - we're checking `if b.selfHostedEtcd` there anyway